docs/topics/cache.txt

```diff
@@ -169,6 +169,127 @@ particularly temporary.
     some problems and seems to be unmaintained. Use ``PyMemcacheCache`` or
     ``PyLibMCCache`` instead.
 
+.. _redis:
+
+Redis
+-----
+
+Redis__ is the most popular cache backend used with Django. It is an in-memory
+database which stores key-value pairs. Redis server can be configured to run as
+a daemon either on the local machine or on a remote instance.
+
+__ https://redis.io/
+
+After setting up the redis server, you'll need to install redis bindings for
+python. `redis-py`_ is the binding supported natively by Django.
+
+.. _`redis-py`: https://pypi.org/project/redis/
+
+To use redis as your cache backend with Django:
+
+* Set :setting:`BACKEND <CACHES-BACKEND>` to
+  ``django.core.cache.backends.redis.RedisCache``
+
+* Set :setting:`LOCATION <CACHES-LOCATION>` to ``ip:port`` values, where ``ip``
+  is the IP address of the redis server and ``port`` is the port on which redis
+  is running, or to a ``unix:path`` value, where ``path`` is the path to a redis
+  Unix socket file.
+
+For example, redis is running on localhost (127.0.0.1) port 6379::
+
+    CACHES = {
+        'default': {
+            'BACKEND': 'django.core.cache.backends.redis.RedisCache',
+            'LOCATION': 'redis://127.0.0.1:6379',
+        }
+    }
+
+Often redis servers are protected with authentication. If order to supply a
+username and password, we add them in the ``LOCATION`` along with the URL::
+
+    CACHES = {
+        'default': {
+            'BACKEND': 'django.core.cache.backends.redis.RedisCache',
+            'LOCATION': 'rediss://username:password@127.0.0.1:6379',
+        }
+    }
+
+If you have multiple redis servers setup in replication mode, you can specify
+the servers in a list. While using multiple servers, all the writes are
+performed on the leader server. The read operation is performed on either of the
+replicas chosen at random. The first server address must be the leader server
+and the other server addresses must be the followers (replicas of leader)::
+
+    CACHES = {
+        'default': {
+            'BACKEND': 'django.core.cache.backends.redis.RedisCache',
+            'LOCATION': [
+                "redis://127.0.0.1:6379", # leader
+                "redis://127.0.0.1:6378", # read-replica 1
+                "redis://127.0.0.1:6377", # read-replica 2
+            ],
+        }
+    }
+
+Along with the standard :ref:`cache_arguments`, the redis caching backend also
+accepts the following additional arguments:
+
+* ``db``: By default, redis ships with 16 logical databases. A database of your
+  choice can be selected either by setting it in the ``LOCATION`` as querystring
+  or path argument, or by setting it in the ``OPTIONS``. The default choice for
+  the database is 0. We can set the database in the following manner::
+
+    CACHES = {
+        'default': {
+            'BACKEND': 'django.core.cache.backends.redis.RedisCache',
+            'LOCATION': 'redis://127.0.0.1:6379?db=10', # option 1
```

Comment:
This looks too much like a full example. However, I think you're saying use on of these three options. I wonder if there's a better way of presenting this.

Reply:
So initially I had three separate examples for the three separate example but the seemed to be too lengthy. Hence I reverted to condensing them into one. Since all three options are not specified in the same manner, I went with the elaborate example. Let me know what you feel!